app.controllers.js

```diff
@@ -12,12 +10,25 @@ const getTopics = (req, res, next) => {
     }
 
 const getArticles = (req, res, next) => {
-    fetchArticles().then((articles) => {
-        res.status(200).send(articles);
-    }).catch(err => {
-        next(err);
-    })
+    const {topic, sort_by, order} = req.query;
+    if (topic) {
+        fetchTopicsByName(topic).then(() => {
```

Comment:
Nice way of creating the error if the topic doesn't exist. Great use of the promise chain in your error handling.